### PR TITLE
Add check_ids API

### PR DESCRIPTION
**Pull request by ytx1991**
### Purpose:
https://github.com/Chia-Network/issue-tracker/issues/332
Add an new API to check asset ownership.

### Current Behavior:
None

### New Behavior:
Call check_ids with a list of IDs and asset type

### Testing Notes:
Unit test

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1334,6 +1335,46 @@ async def verify_signature(self, request) -> EndpointResult:
         else:
             return {"isValid": False, "error": "Signature is invalid."}
 
+    async def check_ids(self, request) -> EndpointResult:
+        """
+        Check if a list of IDs is owned by the current wallet.
+        :param request: {"ids":["ID1", "ID2", ...], "type":"COIN/NFT/DID"}
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:1341]: wouldn't it be easier to provide a list of `(coin_id, coin_type)` vs doing multiple requests depending on the coin type?

File: tests/wallet/rpc/test_wallet_rpc.py [hunk 2]
```diff
@@ -896,7 +896,63 @@ def only_ids(trades):
...
+    nft_id = (await nft_wallet.get_current_nfts())[0].nft_id.hex()
+    res = await wallet_1_rpc.check_ids([nft_id], "NFT")
+    assert len(res) == 1
+    assert res[0] == nft_id
+
+    # COIN
+    coin_id = (await nft_wallet.get_current_nfts())[0].coin.name().hex()
+    res = await wallet_1_rpc.check_ids([coin_id], "COIN")
+    assert len(res) == 1
+    assert res[0] == coin_id
```
Comment (trepca) [inline tests/wallet/rpc/test_wallet_rpc.py:955]: add a test to check if it doesn't return coins that it doesn't own but they should exist

File: chia/rpc/wallet_rpc_api.py [hunk 3]
```diff
@@ -1334,6 +1335,46 @@ async def verify_signature(self, request) -> EndpointResult:
...
+        if "type" not in request:
+            raise ValueError("Type is required by check_ids.")
+        if request["type"].upper() == "COIN":
+            ids: List[bytes32] = [bytes32.fromhex(coin_id) for coin_id in request["ids"]]
+            coin_records: List[
+                WalletCoinRecord
+            ] = await self.service.wallet_state_manager.coin_store.get_multiple_coin_records(ids)
+            valid_ids = [r.coin.name().hex() for r in coin_records]
+        elif request["type"].upper() == "NFT":
+            ids = [
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:1355]: @paninaro how many ids do you plan to check at a time?

Comment (paninaro) [reply chia/rpc/wallet_rpc_api.py:1355]: The primary use case I have in mind would be for checking if NFTs in an offer are owned by the user. We have a cap of ~10 NFTs in an offer when created from the GUI. So I would say ~20 would probably be the max.

File: chia/rpc/wallet_rpc_api.py (hunk 1 above)
Comment (paninaro) [reply chia/rpc/wallet_rpc_api.py:1341]: I agree that being able to query a mix of coin types in a single request would be a nice.

Do we even need to specify the type if we can infer the coin type by the bech32m prefix? `nft1...abcxyz` for NFT, `did:chia:1...def123` for DID, and anything else could be looked up as a coin. Or is there a reason to have the type specified by the caller?

Comment (ytx1991) [reply chia/rpc/wallet_rpc_api.py:1341]: The question is if we want to support hex string for NFT and DID. Also, if the type is a known thing, providing the type will speed up the query.

File: tests/wallet/rpc/test_wallet_rpc.py (hunk 2 above)
Comment (ytx1991) [reply tests/wallet/rpc/test_wallet_rpc.py:955]: I thought only the coins we owned will be saved in the table. What is the reason for saving the coin we don't own?

**Pull request closed without merging**